### PR TITLE
Moved x-axis to bottom of two charts, Moved y-axis outside chart

### LINKED ISSUE #39: Remove Linear Time Series
Use lower level components to compose the charts, which will allow for more flexibility as we add more functionality later on.

### LINKED ISSUE #96: The primary chart and secondary chart should share the same x-axis
The primary chart and secondary chart should share the same x-axis, which should appear below the secondary chart. If the secondary chart is hidden, it should still be shown below the primary chart.

Current:

axis

Desired (rough mock-up):

axis-2

### LINKED ISSUE #101: Display y-axis ticks and close price lozenge to right of primary chart
Display these to right of primary chart to ensure data is not obscured.

### PR DESCRIPTION
**Pull request by djmiley**
Resolves #39. Resolves #96. Partially resolves #101 (displaying y-ticks to right of chart).

### CODE REVIEW

**Comment by tlsim** [comment]
@djmiley Is it possible to fix #96 and #101 but keep linear time series? Lts has an xAxisHeight option and the underlying y-scale `orient` method is rebound as `yOrient`. If not, try using standard d3 components.

**Comment by djmiley** [comment]
WIP

**Comment by djmiley** [comment]
Submitted a pull request to d3fc that should eventually lead to deletion of `sc.chart.lineartimeseries`. Issues with version number mean that I will have to leave it in this request for the moment.

File: src/index.html [hunk 1]
```diff
@@ -59,35 +58,37 @@
...
+        <div class="col-md-12">
+          <svg class="chart x-axis"></svg>
+        </div>
+      </div>        
       <div class="row"> 
         <div class="col-md-12">
           <svg class="chart nav"></svg>
         </div>
-      </div>
-
```
Comment (djmiley) [inline src/index.html]: Accidental deletion, restore this.

File: src/assets/styles/xAxis.less [hunk 2]
```diff
@@ -0,0 +1,4 @@
+.chart.x-axis {
+
+    height: @x-axis-height;
+}
```
Comment (djmiley) [inline src/assets/styles/xAxis.less]: Remove this change

File: src/assets/styles/variables.less [hunk 3]
```diff
@@ -4,6 +4,9 @@
 @annotation-text-colour: #fff;
 @moving-average-line-colour: #f0f;
 
+// Shared X Axis
+@x-axis-height: 20px;
+
```
Comment (djmiley) [inline src/assets/styles/variables.less]: Remove this change

File: src/assets/styles/navigator.less [hunk 4]
```diff
@@ -1,4 +1,4 @@
-.nav {
+.chart.nav {
```
Comment (djmiley) [inline src/assets/styles/navigator.less]: Remove this change

File: src/assets/js/util/calculateDimensions.js [hunk 5]
```diff
@@ -5,12 +5,14 @@
...
+        var headRowHeight = parseInt(container.select('#head-row').style('height'), 10) +
+            parseInt(container.select('#head-row').style('margin-top'), 10) +
+            parseInt(container.select('#head-row').style('margin-bottom'), 10);
         var navHeight = parseInt(container.select('svg.nav').style('height'), 10);
+        var xAxisHeight = parseInt(container.select('svg.x-axis').style('height'), 10);
 
         var useableScreenWidth = parseInt(container.style('width'), 10) - (leftPadding + rightPadding);
-        var useableScreenHeight = window.innerHeight - headRowHeight - navHeight -
-            2 * fc.chart.linearTimeSeries().xAxisHeight();
+        var useableScreenHeight = window.innerHeight - headRowHeight - navHeight - xAxisHeight;
```
Comment (djmiley) [inline src/assets/js/util/calculateDimensions.js]: Put into separate bug fixing pull request

**Comment by djmiley** [comment]
I'm going to close this for now in light of #172

**Pull request closed without merging**